File: 1544-count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void func(TreeNode* root,int pathMax,int& cnt){
        if(root==nullptr) return;
        if(root->val>=pathMax){
            cnt++;
            pathMax=root->val;
        } 
        func(root->left,pathMax,cnt);
        func(root->right,pathMax,cnt);
    }
    int goodNodes(TreeNode* root) {
        int cnt=0;
        func(root,INT_MIN,cnt);
    return cnt;
    }
};

File: 1544-count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    void dfs(TreeNode node ,int pathMax,int[] cnt){
        if(node==null) return;
        if(node.val>=pathMax){
            cnt[0]++;
            pathMax=node.val;
        } 
        dfs(node.left,pathMax,cnt);
        dfs(node.right,pathMax,cnt);
    }
    public int goodNodes(TreeNode root) {
        int[] cnt=new int[1];
        dfs(root,Integer.MIN_VALUE,cnt);
        return cnt[0];
    }
}